### ISSUE TITLE
REST API not running

### ISSUE THREAD

**Issue by codekitchen** [openhab/openhab#1906]
Running openhab 1.6.1 on an Intel Edison board, which is a 32-bit x86 embedded environment.

Everything seems to be working properly, except the REST API is not running -- visiting the `/rest` URL or any URL under that results in a 404 error `Problem accessing /rest. Reason: ProxyServlet: /rest`.

I turned logging up to the DEBUG level, but the only relevant information I can see is that "Started REST API at /rest" is logged multiple times during startup. Interestingly, if I access /rest while openhab is starting up, I can hit the API, but then if I retry a few seconds later, the API is gone and I get that 404 error.

I imagine it has to be some sort of error during startup, but without any other logging information available, I'm not sure what to do next to track down the issue.

```
2014-12-15 09:20:47.028 [INFO ] [j.s.i.a.DeferredResourceConfig] - Instantiated the Application class org.openhab.io.rest.RESTApplication
2014-12-15 09:20:48.473 [INFO ] [penhab.io.rest.RESTApplication] - Started REST API at /rest
2014-12-15 09:20:48.520 [INFO ] [cpr.DefaultAnnotationProcessor] - AnnotationProcessor class org.atmosphere.cpr.DefaultAnnotationProcessor$
BytecodeBasedAnnotationProcessor being used
2014-12-15 09:20:48.573 [INFO ] [.a.h.ReflectorServletProcessor] - Installing Servlet com.sun.jersey.spi.container.servlet.ServletContainer
2014-12-15 09:20:48.579 [INFO ] [c.s.j.s.i.a.WebApplicationImpl] - Initiating Jersey application, version 'Jersey: 1.17.1 02/28/2013 12:47
PM'
2014-12-15 09:20:48.582 [INFO ] [c.s.j.s.i.a.WebApplicationImpl] - Adding the following classes declared in META-INF/services/jersey-server
-components to the resource configuration:
  class org.atmosphere.jersey.AtmosphereResourceConfigurator
2014-12-15 09:20:48.643 [INFO ] [j.s.i.a.DeferredResourceConfig] - Instantiated the Application class org.openhab.io.rest.RESTApplication
2014-12-15 09:20:50.051 [INFO ] [penhab.io.rest.RESTApplication] - Started REST API at /rest
2014-12-15 09:20:50.138 [INFO ] [cpr.DefaultAnnotationProcessor] - AnnotationProcessor class org.atmosphere.cpr.DefaultAnnotationProcessor$
BytecodeBasedAnnotationProcessor being used
2014-12-15 09:20:50.220 [INFO ] [.a.h.ReflectorServletProcessor] - Installing Servlet com.sun.jersey.spi.container.servlet.ServletContainer
2014-12-15 09:20:50.226 [INFO ] [c.s.j.s.i.a.WebApplicationImpl] - Initiating Jersey application, version 'Jersey: 1.17.1 02/28/2013 12:47
PM'
2014-12-15 09:20:50.229 [INFO ] [c.s.j.s.i.a.WebApplicationImpl] - Adding the following classes declared in META-INF/services/jersey-server
-components to the resource configuration:
  class org.atmosphere.jersey.AtmosphereResourceConfigurator
```

Java version:

```
java version "1.7.0_67"
Java(TM) SE Runtime Environment (build 1.7.0_67-b01)
Java HotSpot(TM) Client VM (build 24.65-b04, mixed mode)
```

I'd love to help track this down, but I could use some pointers if anybody has any. I'm a developer, but I'm not very familiar with Java or the JVM.

**Comment by teichsta** [comment]
Could you please make a clean startup using the start_debug.xxx script? Are you using the unchanged 1.6.1 Release Binary?

**Comment by codekitchen** [comment]
Yep, unmodified 1.6.1. I didn't get any errors logged using the start_debug.sh script, but I happened to start poking around when I saw some logging about mdns, and discovered that the issue is related to the system mdns service that runs by default on this Intel Edison system. It appears that starting up the REST API silently fails partway through the process, if another application is already providing mdns services, and thus listening on the mdns UDP port (5353 I believe).

If I disable the system mdns service (with `systemctl stop mdns` for anybody else using this board) and restart openhab, the REST API is available. I toggled it a couple times and restarted openhab each time, to be sure, and can say with pretty high confidence that this is the issue.

Is there a setting available that would tell openhab to skip starting its internal mdns service?

**Comment by dominicdesu** [comment]
Hi, I don't know if this is related but cannot hurt to mention it:
There seems to be a problem when running openHAB with activated maxcube and REST binding from eclipse because JmDNS isn't initialized correctly. See here for details: https://groups.google.com/d/topic/openhab/PAnKhrsDqjU/discussion

**Comment by codekitchen** [comment]
I'm not sure if it's related either, but I don't have the maxcube binding activated. The addons I have: networkhealth, weather, zwave, and rrd4j.

**Comment by teichsta** [comment]
Thanks @codekitchen for investigating this!

> Is there a setting available that would tell openhab to skip starting its internal mdns service?

no, not yet!

As a workaround you could stop the ```org.openhab.io.servicediscovery``` Bundle through the OSGi console or remove the Binary (jar) from your installation.

Since this doesn't seem to be Bug but rather a feature request (please make ServiceDiscovery Start configurable) i'd like to close this issue. Would you mind entering a Feature Request which refers to this Issue?

Thanks, Thomas E.-E.

**Comment by teichsta** [comment]
@kaikreuzer this would also be relevant for ESH …

**Comment by kaikreuzer** [comment]
Thanks for the notice, this has been addressed with https://github.com/eclipse/smarthome/pull/146

**Comment by fjfo** [comment]
I got the same problem with openHAB runtime v1.7.1.
- everything is good,
- webapp at openhab.app is ok in HTTP/HTTPS
- Remove org.openhab.io.servicediscovery.jar in server/plugins
BUT when i query: http://myip:8080/rest/
nothing shown up.

Here my STDOUT when I ran with start_debug.sh
./start_debug.sh
Launching the openHAB runtime in debug mode...
Listening for transport dt_socket at address: 8001
osgi> 23:20:29.543 [DEBUG] [.s.internal.SchedulerActivator:36   ] - Scheduler has been started.
23:20:29.720 [INFO ] [uartz.impl.StdSchedulerFactory:1175 ] - Using default implementation for ThreadExecutor
23:20:29.746 [INFO ] [rtz.core.SchedulerSignalerImpl:61   ] - Initialized Scheduler Signaller of type: class org.quartz.core.SchedulerSignalerImpl
23:20:29.747 [INFO ] [rg.quartz.core.QuartzScheduler:243  ] - Quartz Scheduler v.2.1.7 created.
23:20:29.749 [INFO ] [org.quartz.simpl.RAMJobStore  :154  ] - RAMJobStore initialized.
23:20:29.751 [INFO ] [rg.quartz.core.QuartzScheduler:268  ] - Scheduler meta-data: Quartz Scheduler (v2.1.7) 'openHAB-job-scheduler' with instanceId 'NON_CLUSTERED'
  Scheduler class: 'org.quartz.core.QuartzScheduler' - running locally.
  NOT STARTED.
  Currently in standby mode.
  Number of jobs executed: 0
  Using thread pool 'org.quartz.simpl.SimpleThreadPool' - with 2 threads.
  Using job-store 'org.quartz.simpl.RAMJobStore' - which does not support persistence. and is not clustered.

23:20:29.751 [INFO ] [uartz.impl.StdSchedulerFactory:1324 ] - Quartz scheduler 'openHAB-job-scheduler' initialized from specified file: './etc/quartz.properties'
23:20:29.752 [INFO ] [uartz.impl.StdSchedulerFactory:1328 ] - Quartz scheduler version: 2.1.7
23:20:29.752 [INFO ] [rg.quartz.core.QuartzScheduler:534  ] - Scheduler openHAB-job-scheduler_$_NON_CLUSTERED started.
23:20:29.756 [DEBUG] [o.config.core.ConfigDispatcher:146  ] - Processing openHAB default configuration file '/mnt/openhab/configurations/openhab_default.cfg'.
23:20:29.758 [DEBUG] [o.config.core.ConfigDispatcher:168  ] - Processing openHAB main configuration file '/mnt/openhab/configurations/openhab.cfg'.
23:20:29.922 [DEBUG] [.o.core.internal.CoreActivator:93   ] - UUID file already exists at '/mnt/openhab/webapps/static/uuid' with content '6e6a4191-d323-4658-927c-7345ca880368'
23:20:29.923 [DEBUG] [.o.core.internal.CoreActivator:115  ] - Created file '/mnt/openhab/webapps/static/version' with content '1.7.1'
23:20:29.924 [INFO ] [.o.core.internal.CoreActivator:61   ] - openHAB runtime has been started (v1.7.1).
23:20:29.931 [DEBUG] [o.config.core.ConfigDispatcher:271  ] - Scheduled refresh job 'ConfigDispatcher.Refresh' in DefaulScheduler
23:20:29.981 [DEBUG] [a.internal.AutoUpdateActivator:31   ] - AutoUpdate binding has been started.
23:20:31.110 [DEBUG] [.p.i.PersistenceModelActivator:23   ] - Registered 'persistence' configuration parser
23:20:31.142 [DEBUG] [.c.t.i.TransformationActivator:38   ] - Transformation Service has been started.
23:20:31.246 [DEBUG] [m.internal.MultimediaActivator:34   ] - Multimedia I/O bundle has been started.
23:20:31.281 [DEBUG] [.io.transport.mqtt.MqttService:123  ] - Starting MQTT Service...
23:20:31.424 [DEBUG] [.i.internal.ItemModelActivator:24   ] - Registered 'item' configuration parser
23:20:31.572 [DEBUG] [o.o.c.i.items.ItemRegistryImpl:137  ] - Item provider 'GenericItemProvider' has been added.
23:20:32.769 [DEBUG] [.o.m.s.i.SitemapModelActivator:23   ] - Registered 'sitemap' configuration parser
23:20:32.900 [INFO ] [.o.u.w.i.servlet.WebAppServlet:79   ] - Started Classic UI at /openhab.app
23:20:33.188 [DEBUG] [.r.internal.RuleModelActivator:42   ] - Registered 'rules' configuration parser
23:20:33.222 [DEBUG] [m.r.internal.engine.RuleEngine:77   ] - Started rule engine
23:20:36.125 [DEBUG] [.s.internal.SamsungTvActivator:34   ] - Samsung TV binding has been started.
23:20:36.158 [DEBUG] [.b.exec.internal.ExecActivator:34   ] - Exec binding has been started.
23:20:36.189 [INFO ] [.service.AbstractActiveService:169  ] - Exec Refresh Service has been started
23:20:40.838 [INFO ] [c.internal.ModelRepositoryImpl:80   ] - Loading model 'default.sitemap'
23:20:41.083 [INFO ] [c.internal.ModelRepositoryImpl:80   ] - Loading model 'default.items'
23:20:41.104 [DEBUG] [i.internal.GenericItemProvider:154  ] - Processing binding configs for items from model 'default.items'
23:20:41.113 [DEBUG] [i.internal.GenericItemProvider:133  ] - Read items from model 'default.items'

Any help?
